week_1/parseutil.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+)
+
+func parseInput(input string) (float64, float64, string, error) {
+	operatorIndex := strings.IndexAny(input, "+-*/")
+
+	if operatorIndex == -1 {
+		err := errors.New("Missing Operator")
+		return 0, 0, "", err
+	}
+
+	expr := strings.Split(input, input[operatorIndex:operatorIndex+1])
+	if len(expr) != 2 {
+		err := errors.New("The number argument of input is invalid")
+		return 0, 0, "", err
+	}
+
+	first := strings.Replace(expr[0], " ", "", -1)
```

Comment:
Check first number and second number are duplicated code, better split to separate function to reuse and enhance readibility